packages/muelu/test/unit_tests/BlockedRepartition.cpp

```diff
@@ -938,14 +941,331 @@ namespace MueLuTests {
   } // BlockedRAPFactoryWithDiagonal
 #endif
 
+  TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(BlockedRepartition, BlockedRAPFactoryCoords, Scalar, LocalOrdinal, GlobalOrdinal, Node)
+  {
+#   include "MueLu_UseShortNames.hpp"
+  MUELU_TESTING_SET_OSTREAM;
+  MUELU_TESTING_LIMIT_SCOPE(Scalar,GlobalOrdinal,NO);
+  out << "version: " << MueLu::Version() << std::endl;
 
+  RCP<const Teuchos::Comm<int> > comm = TestHelpers::Parameters::getDefaultComm();
 
-#if defined(HAVE_MUELU_ISORROPIA) && defined(HAVE_MUELU_ZOLTAN)
+  if(comm->getSize() == 1) {
+    out << "Skip BlockedRepartion test in serial case" << std::endl;
+    return;
+  }
+
+  /**********************************************************************************/
+  /* CREATE INITIAL MATRIX                                                          */
+  /**********************************************************************************/
+  RCP<const Map> bigMap;
+  RCP<const Map> map1;
+  RCP<const Map> map2;
+  GO numElements = 400;
+  GO numElements1 = 200;
+  GO numElements2 = 200;
```

Comment:
@pohm01 Are the blocks allowed to have different numbers of elements?  If so, would the test be more general and/or exercise more parts of the blocked transfer factory?  It's fine if the answer is "no", I'm just wondering.

Reply:
Currently there is only support for one set of coordinates that is shared between the blocks, so the blocks need the same number of elements and co-located dofs/nodes.